kcidb/monitor/test_output.py

```diff
@@ -85,3 +87,123 @@ def test_min(self):
+                        "rdma-dev-team@redhat.com"
+                    ],
+                    "start_time": "2020-03-02T15:16:15.790000+00:00",
+                    "git_repository_branch": "wip/jgg-for-next",
+                    "git_commit_hash": "5e29d1443c46b6ca70a4c940a67e8c09f05dcb7e",
+                    "patchset_hash": "",
+                    "git_repository_url": "https://git.kernel.org/pub/scm/linux/kernel/git/arm64/linux.git",
+                    "misc": {
+                        "pipeline_id": 467715
+                    },
+                    "id": "origin:1",
+                    "origin": "origin",
+                    "patchset_files": [],
+                    "valid": True,
+                },
+            ],
+            "builds": [
+                                {
+                    "id": "kernelci:kernelci.org:619fecfc6d932a49b5f2efb0",
+                    "checkout_id": "origin:1",
+                    "origin": "origin",
+                    "comment": "v5.16-rc2-19-g383a44aec91c",
+                    "start_time": "2021-11-25T20:07:24.499000+00:00",
+                    "duration": 486.695294857,
+                    "architecture": "x86_64",
+                    "config_name": "x86_64_defconfig+x86-chromebook+kselftest",
+                    "valid": True
+                },
+            ]
+        })
+        revision = oo_client.query(orm.Pattern.parse(">revision#"))["revision"][0]
+
+        # test under-limit length subject/body are left intact in the email
+        message = NotificationMessage(
+            ["kernelci-results-staging@groups.io"],
+            subject="Test completed for linux.git",
+            body="Below is the summary of results Kernel CI database has "
+                 "recorded for this revision so far")
+        notification = Notification(revision, "test_subscription", message)
+        message = notification.render()
+        self.assertIsInstance(message, email.message.EmailMessage)
+        self.assertEqual(len(message.get("Subject")), 28)
+        self.assertEqual(len(message.get_body('plain').get_content()[:-1]), 88)
+
+        # test zero length subject/body produces zero length subject and body in the email
+        empty_message = NotificationMessage(
+            ["kernelci-results-staging@groups.io"],
+            subject="", body="")
+        empty_notification = Notification(revision, "test_subscription_2", empty_message)
+        empty_message = empty_notification.render()
+        self.assertEqual(len(empty_message.get("Subject")), 0)
+        self.assertEqual(len(empty_message.get_body('plain').get_content()[:-1]), 0)
+
+        # test exact-limit length subject/body are left intact in the email,
+        exact_message = NotificationMessage(
+            ["kernelci-results-staging@groups.io"],
+            subject="Test completed for linux.git:123" * 8,
+            body="Below is the summary of results Kernel CI database has recorded " * 64
```

Comment:
Could you generate the input strings using the `*_MAX_LEN` constants as well, here and below? Something like `subject="S" * NOTIFICATION_MESSAGE_SUBJECT_MAX_LEN, body="B" * NOTIFICATION_MESSAGE_BODY_MAX_LEN`?